AWSS3/AWSS3TransferUtility.m

```diff
@@ -325,8 +332,10 @@ + (void)registerS3TransferUtilityWithConfiguration:(AWSServiceConfiguration *)co
                                                                                      recoverState:NO
                                                                                 completionHandler:completionHandler];
     if (s3TransferUtility) {
+        NSAssert(_serviceClients != nil, @"Value is required");
```

Comment:
We should be using `NSCAssert` instead of `NSAssert` since it does not work with ARC.